dashboard/src/actions/endpointAction.js

```diff
@@ -0,0 +1,25 @@
+import * as TYPES from "./types";
+
+export const fetchEndpoints = () => async (dispatch) => {
+  try {
+    const response = await fetch("/api/v1/endpoints");
+
+    if (!response.ok) {
+      throw new Error(`HTTP error! status: ${response.status}`);
+    }
+    const data = await response.json();
+    window.endpoints = data;
+    dispatch({
+      type: TYPES.SET_ENDPOINTS,
+      payload: { ...data },
+    });
+  } catch (error) {
+    dispatch({
+      type: TYPES.SHOW_TOAST,
+      payload: {
+        variant: "danger",
+        title: "Something went wrong, try again later",
+      },
+    });
```

Comment:
"Something went wrong", but we're not going to give any hint as to what??  (Can't we string-ize `error` and include it in the message?  That would be much more useful than the suggestion to "try again later"....)